llvm/lib/Target/AArch64/AArch64.td

```diff
@@ -179,3 +179,21 @@ def AArch64 : Target {
 //===----------------------------------------------------------------------===//
 
 include "AArch64PfmCounters.td"
+
+
+//===----------------------------------------------------------------------===//
+// GlobalISel patterns
+//===----------------------------------------------------------------------===//
+
+include "AArch64GlobalISelPatterns.td"
+
+// We want to first hit the instruction patterns.
+foreach VT = [nxv2bf16, nxv4bf16, nxv8bf16] in {
```

Comment:
Having this in the top level .td file seems like bad file structure.

Reply:
Yeah maybe. It is actually a cheap trick. It was interfering with the new fabs and fneg patterns. We have now patterns and can use the instead of and and xor. If this is the second to last issue, I am more than happy to move them.

Reply:
Upstream they are in AArch64SVEInstrInfo.td. I just need to find a place for the new patterns to be before the old patterns.

https://github.com/llvm/llvm-project/blob/5fae408d3a4c073ee43aec9906fa44ffe4026301/llvm/lib/Target/AArch64/AArch64SVEInstrInfo.td#L686

Reply:
I put the GlobalIsel include at the bottom of the file to minimize it's priority and don't bother others. It depends on AArch64SVEInstrInfo.td for the definition of the fabs and fneg instructions. At the same time, the and and xor patterns for fabs and fneg are defined in AArch64SVEInstrInfo.td. They have higher priority than my patterns.

Reply:
I don't understand why these bf16 patterns are interfering with your new patterns that don't mention bf16.